src/agent.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { Finding, FindingSeverity, FindingType, HandleTransaction, ethers } from 'forta-agent';
+import { createAddress } from 'forta-agent-tools';
+import { TestTransactionEvent, MockEthersProvider } from 'forta-agent-tools/lib/test';
+import { provideHandleTransaction } from './agent';
+import { swapEvent, uniswapPoolABI, UNI_INIT_CODE, UNISWAP_FACTORY_ADDRESS } from './agent'; // Import ABI and constants
+import { computePoolAddress } from './agent'; // Import computeAddress function
+
+describe('Uniswap V3 Swap Event Detector', () => {
+  let handleTransaction: HandleTransaction;
+  let Iface: ethers.utils.Interface;
+
+  const mockFactoryAddress = createAddress(UNISWAP_FACTORY_ADDRESS);
```

Comment:
As you're correctly passing the factory address as a parameter to the `provideHandleTransaction`, I would sugges to use a mock address here (so that the tests prove the logic is working fine independently of the factory)